Common/include/basic_types/datatype_structure.hpp

```diff
@@ -41,130 +41,132 @@
-  FORCEINLINE const T& _printGetValue(const T& val) {return val;}
-  /*--- Overload for expressions of active types. ---*/
-  template<class T, su2enable_if<!std::is_trivial<T>::value> = 0>
-  FORCEINLINE passivedouble _printGetValue(const T& val) { return val.getValue(); }
-
-  /*!
-   * \brief Wrapper to sprintf to be able to print active types and AD expressions.
-   * \note This is for compatibility with old code, stringstreams should be the preferred way to build strings.
-   * \param[in] str - Target char buffer.
-   * \param[in] format - Format string.
-   * \param[in] args - Values to be printed to the string.
-   */
-  template<class... Ts>
-  FORCEINLINE void sprintf(char* str, const char* format, Ts&&... args) {
-    ::sprintf(str, format, SU2_TYPE::_printGetValue(args)...);
-  }
-  FORCEINLINE void sprintf(char* str, const char* literal) {
-    ::sprintf(str, "%s", literal);
-  }
+/*!
+ * \brief Get the passive value of any variable. For most types return directly,
+ * specialize for su2double to call GetValue.
+ * \note This is a struct instead of a function because the return type of the
+ * su2double specialization changes.
+ */
+template <class T>
+struct Passive {
+  FORCEINLINE static T Value(const T& val) { return val; }
+};
+template <>
+struct Passive<su2double> {
+  FORCEINLINE static passivedouble Value(const su2double& val) { return GetValue(val); }
+};
+
+/*!
+ * \brief Casts the primitive value to int (uses GetValue, already implemented for each type).
+ * \param[in] data - The non-primitive datatype.
+ * \return - The primary value casted to int.
+ */
+FORCEINLINE int Int(const su2double& data) { return static_cast<int>(SU2_TYPE::GetValue(data)); }
+
+/*!
+ * \brief Casts the primitive value to short (uses GetValue, already implemented for each type).
+ * \param[in] data - The non-primitive datatype.
+ * \return - The primary value casted to short.
+ */
+FORCEINLINE short Short(const su2double& data) { return static_cast<short>(SU2_TYPE::GetValue(data)); }
+
+/*--- Special handling of the sprintf routine for non-primitive types. ---*/
+/*--- Pass-through for built-in types. ---*/
+template <class T, su2enable_if<std::is_trivial<T>::value> = 0>
+FORCEINLINE const T& _printGetValue(const T& val) {
+  return val;
+}
+/*--- Overload for expressions of active types. ---*/
+template <class T, su2enable_if<!std::is_trivial<T>::value> = 0>
+FORCEINLINE passivedouble _printGetValue(const T& val) {
+  return val.getValue();
+}
+
+/*!
+ * \brief Wrapper to sprintf to be able to print active types and AD expressions.
+ * \note This is for compatibility with old code, stringstreams should be the preferred way to build strings.
+ * \param[in] str - Target char buffer.
+ * \param[in] format - Format string.
+ * \param[in] args - Values to be printed to the string.
+ */
+template <class... Ts>
+FORCEINLINE void sprintf(char* str, const char* format, Ts&&... args) {
+  ::sprintf(str, format, SU2_TYPE::_printGetValue(args)...);
```

Comment:
> Thanks. Is it possible to target only some paths for formatting? I'm concerned about making life too difficult for developers, and we cannot break lines in some files because we parse them to generate some python code. Having only some paths we would then include more files over time until we cover everything.

AFAIK targeting specific paths are not possible, it works the other way around we can exclude some paths. Here I only exclude the `external` directory.

The hook options of pre-commit hooks are [here](https://pre-commit.com/index.html#pre-commit-configyaml---hooks) 

I look at the regression test the values seem okay but the test fails is this due to wrappers? If you tell me which directories should be omitted, I would adjust it quickly. In general, we can also disable column limit/wrapping from `clang-format` but it is probably not the best option.